blob/s3blob/s3blob.go

```diff
@@ -134,19 +134,25 @@ func (o *URLOpener) OpenBucketURL(ctx context.Context, u *url.URL) (*blob.Bucket
 }
 
 // Options sets options for constructing a *blob.Bucket backed by fileblob.
-type Options struct{}
+type Options struct {
+	UseLegacyListObjects bool
```

Comment:
If I understand correctly, S3 itself should support both modes, right?

If so, I'd like to make the tests for S3 run in both modes so we have coverage of both paths. I could do this as a follow-up on PR from yours.

Reply:
I verified that this works (i.e., I ran all of the conformance tests for S3 using `UseLegacyListObjects=true` and they passed). So, resolving this comment.